include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -407,7 +407,7 @@ inline void format_sam::read_alignment_record(stream_type & stream,
                   "The ref_offset must be a specialisation of std::optional.");
 
     auto stream_view = views::istreambuf(stream);
-    auto field_view = stream_view | views::take_until_or_throw_and_consume(is_char<'\t'>);
+    auto field_view = stream_view | views::take_until_or_throw(is_char<'\t'>);
```

Comment:
Why this change?

Reply:
So the initial problem was that `*` indicates and empty SAM field and the function `read_field` only checks if `*` is present (before this PR). But I did not check if the field ONLY contains the `*`. 

So I need to check if there is a `\t` after the `*`but I cannot do this if the `\t` is consumed in general (the change you asked the reason for). Note that I cannot check if `field_view` is empty instead of checking if the next char is `\t` because on an input_range the `take..` view will always "regenerate".

```cpp
// input_range = "1\t2\t3\n"
auto v = input_range | take_until_and_consume('\t');
seqan3::debug_stream << v << std::endl; // prints 1
seqan3::debug_stream << v << std::endl;  // prints 2
```

Is that understandable?

Reply:
Yep; unfortunate that we can't keep the consume :(

Reply:
This is only problematic if there are empty fields / double tabs. This is not allowed in SAM/BAM.

Note that in your example `v`is "at-end" in between the calls. The second (implicit) call to to `begin()` resets it. As long as you don't do that, there is no problem.

Reply:
> Note that in your example vis "at-end" in between the calls, i.e. the first begin iterator does compare true to the end. Otherwise it wouldn't stop printing after "1". Only when you create a new iterator (which implicitly happens in your example) does this new iterator start at the next field.

Ah ok thanks. That's the fact that I was missing when trying to fix this :)